src/ipa/rpi/cam_helper/cam_helper_imx335.cpp

```diff
@@ -0,0 +1,69 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright (C) 2023, Ideas on Board Oy.
+ *
+ * cam_helper_Imx335.cpp - camera information for Imx335 sensor
+ */
+
+#include <assert.h>
+
+#include "cam_helper.h"
+#include "math.h"
+using namespace RPiController;
+
+class CamHelperImx335 : public CamHelper
+{
+public:
+	CamHelperImx335();
+	uint32_t gainCode(double gain) const override;
+	double gain(uint32_t gainCode) const override;
+	unsigned int hideFramesModeSwitch() const override;
+
+private:
+	/*
+	 * Smallest difference between the frame length and integration time,
+	 * in units of lines.
+	 */
+	static constexpr int frameIntegrationDiff = 4;
+	static constexpr uint32_t maxGainCode = 240;
```

Comment:
https://github.com/torvalds/linux/blob/master/drivers/media/i2c/imx335.c#L55
`#define IMX335_AGAIN_MAX		100`

Reply:
Aha - wow - that's old references indeed ... The helpers must be 2 years old and never reviewed :-( sorry!